src/robotFactory.js

```diff
@@ -1,10 +1,63 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.coords.x = coords.x ? coords.x : 0;
+    this.coords.y = coords.y ? coords.y : 0;
```

Comment:
Please improve your code style according this comment not only in these line but on whole your solution

```suggestion
    this.coords.x = coords.x || 0;
    this.coords.y = coords.y || 0;
```